site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -665,22 +723,46 @@ def transform_pids(self):
 
     def transform_record(self):
         """Perform record transformation."""
+        inspire_id = self.inspire_metadata.get("control_number")
+        current_app.logger.debug(f"[inspire_id={inspire_id}] Starting transform_record")
+
+        current_app.logger.debug(f"[inspire_id={inspire_id}] Transforming metadata")
+        metadata = self.transform_metadata()
```

Comment:
These APIs remind me the importer: the danger is that there is no enforced order of calling the various "sub-methods", and they might also be using/setting `self`.

Can we instead have a `transform` method or similar, and then delegate the "sub-transformation" to another object?
E.g.
```
def transform_record(inspire_record):  # can we avoid `self` here?
    ...
    transformed = RecordTransformer().transform()
    transformed.metadata
    transformed.custom_fileds
    ...
```
just an example